File: CS/WindowsApplication3/Custom Ribbon/MyRibbonControl.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraBars.Ribbon;

namespace DXSample {
    public class MyRibbonControl : RibbonControl
    {
        public MyRibbonControl() : base() { }
        protected override DevExpress.XtraBars.Ribbon.ViewInfo.RibbonViewInfo CreateViewInfo()
        {
            return new MyRibbonViewInfo(this);
        }
        protected override RibbonBarManager CreateBarManager()
        {
            return new MyRibbonBarManager(this);
        }

        public event CustomDrawBarItemEventHandler CustomDrawBarItemLink;
        internal void RaiseCustomDrawBarItemLink(CustomDrawBarItemEventArgs e)
        {
            if (CustomDrawBarItemLink != null)
                CustomDrawBarItemLink(this, e);
        }
    }
}

File: VB/WindowsApplication3/Custom Ribbon/MyRibbonControl.vb
Imports DevExpress.XtraBars.Ribbon

Namespace DXSample

    Public Class MyRibbonControl
        Inherits RibbonControl

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Function CreateViewInfo() As ViewInfo.RibbonViewInfo
            Return New MyRibbonViewInfo(Me)
        End Function

        Protected Overrides Function CreateBarManager() As RibbonBarManager
            Return New MyRibbonBarManager(Me)
        End Function

        Public Event CustomDrawBarItemLink As CustomDrawBarItemEventHandler

        Friend Sub RaiseCustomDrawBarItemLink(ByVal e As CustomDrawBarItemEventArgs)
            RaiseEvent CustomDrawBarItemLink(Me, e)
        End Sub
    End Class
End Namespace
